api/object.go

```diff
@@ -301,6 +355,16 @@ func (opts DeleteObjectOptions) Apply(values url.Values) {
 	}
 }
 
+func (opts HeadObjectOptions) ApplyHeaders(h http.Header) {
```

Comment:
Similar thing here. The endpoints are bound to be the same so maybe it's fine to reuse the `Options` from the endpoint we already have?